Simplify some build infrastructure

dougbu opened the pull request:
- fix Interop test asset projects
  - use normal mechanism to target latest ASP.NET Core bits
- reduce duplicate shared Fx project references
  - ensure App.Ref is built when targeting packs are building
  - ensure shared Fx is built before creating correlation payload
  - remove (unusual) infrastructure only used in `InteropTests`
- !wip! Reduce build steps in our pipelines
  - `msbuild` now enforces the ordering as needed for tests
  - do not build Java projects in quarantined tests

nit: be explicit and use `-arch x64` in affected build steps that aren't for ARM64

dougbu committed:
Make Interop test asset projects more "normal"
- use normal mechanism to target latest ASP.NET Core bits
- remove unusual infrastructure that was only used in `InteropTests`
- remove use of `$(BuildInteropProjects)`

dougbu committed:
Increase shared Fx project reference consistency
- ensure App.Runtime and App.Ref (if needed) packages are built before creating correlation payload,
  before testing locally, and before building Interop test assets
  - check `$(IsHelixJob)` to slightly reduce the number of App.Ref / App.Runtime references
    - set property in RunHelix.ps1 too
  - only the shared Fx / targeting pack layouts are needed in some cases but always pack for further consistency
- move `$(IsTargetingPackBuilding)` to Versions.props; make property available in helix.proj

nits:
- add a few more comments about targeting pack to Versions.props
- modernize documentation in the project template README

dougbu committed:
Reduce build steps in our pipelines
- `msbuild` now enforces the ordering as needed for tests
- do not build Java projects in quarantined tests

nits:
- be explicit and use `-arch x64` in affected build steps
- remove `/p:BuildInteropProjects=true /p:RunTemplateTests=true` because an earlier commit removed
`$(BuildInteropProjects)` and `$(RunTemplateTests)` defaults to `true`

dougbu commented:
Oops, this is just a draft. Ignore for now&hellip;

AraHaan commented on eng/helix/helix.proj:
```diff
@@ -12,6 +12,20 @@
 
+  <!-- Ensure shared framework is available for correlation payload. -->
+  <ItemGroup Condition=" '$(DoNotRequireSharedFxHelix)' != 'true' " >
+    <ProjectReference Include="$(RepoRoot)\src\Framework\App.Ref\src\Microsoft.AspNetCore.App.Ref.csproj"
```
I wonder why these are not referenced in the wixproj's for the installers instead of it trying to (and failing) to unzip the files to a zip file that does not exists (which I think could be avoided if they referenced these as well).

dougbu committed:
Require ARM64 build tools and ATL components
- include when using InstallVisualStudio.ps1
- fortunately, components are available in both VS 2019 and VS 2022

dougbu committed:
Limit Helix operations
- deal w/ repeated builds for Microsoft.DotNet.Helix.Sdk.MultiQueue.targets and ...MonoQueue.targets
- in outer build (the one that executes inner builds then waits for Helix work items), do very little
- in first inner build, create ASP.NET correlation payload and publish all test projects
- in all inner build(s), create and submit per-queue work items

nits:
- remove `$(HelixTestConfigurationFilePath)` and related `@(HelixCorrelationPayload)` setting that Helix SDK does
- use `<Choose/>` in one place to make either / or properties more obvious
